### PR TITLE
Ed 1068 external supplier proxy view

### PR DESCRIPTION
**Pull request by sdonk**
I added pytest sugar as well

image

### CODE REVIEW

**Comment by konradko** [comment]
```Invalid requirement: 'pytest-sugar=0.8.0'
= is not a valid operator. Did you mean == ?```

**Comment by richtier** [comment]
that's some very nice sugar :)

File: company/tests/test_proxy.py [hunk 1]
```diff
@@ -1,53 +1,70 @@
...
+)
 
-    url = reverse('api-external-company', kwargs={'sso_id': 2})
 
-    response = client.get(url)
+@pytest.mark.parametrize(('view_name',), data, ids=ids)
+def test_proxy_api_view_bad_signature(view_name, client):
+    with patch(
+            'ui.signature.external_api_checker.test_signature'
+    ) as mock_test_signature:
```
Comment (richtier) [inline company/tests/test_proxy.py]: why this change? it reduces the readability of the code.

does `patch` not work as a decorator when `@pytest.mark.parametrize` is in play?

File: company/tests/test_proxy.py [hunk 2]
```diff
@@ -1,53 +1,70 @@
...
-    mock_test_signature.return_value = False
+ids = (
+    'external company view',
+    'external supplier view'
+)
 
-    url = reverse('api-external-company', kwargs={'sso_id': 2})
 
-    response = client.get(url)
+@pytest.mark.parametrize(('view_name',), data, ids=ids)
```
Comment (richtier) [inline company/tests/test_proxy.py]: cool feature :+1: . could you rename `data` to something more specific such as `view_names` please. similar comment for `ids`.

File: company/tests/test_proxy.py (hunk 1 above)
Comment (sdonk) [inline company/tests/test_proxy.py]: I know it's not ideal, but I have never managed to have `patch` and `parametrize` working together as decorators. The parametrize magic breaks the other decorator(s).

File: company/tests/test_proxy.py (hunk 2 above)
Comment (sdonk) [inline company/tests/test_proxy.py]: Changed

File: company/tests/test_proxy.py (hunk 1 above)
Comment (sdonk) [inline company/tests/test_proxy.py]: @richtier I can use a different syntax for the parametrized tests (the `yield` one used by `nose`, `pytest` can run it), if you prefer to keep `patch` as decorators

Comment (richtier) [inline company/tests/test_proxy.py]: great :)

Comment (richtier) [inline company/tests/test_proxy.py]: no worries. readability counts, but if there is a technical constraint then no worries. LGTM.
